new_array_processing.py

```diff
@@ -84,24 +83,26 @@ def get_volcano_backazimuth(latlist, lonlist, volcanoes):
             print('Reading in data from IRIS')
             client = Client("IRIS")
             try:
-
                 st = client.get_waveforms(NET, STA, LOC, CHAN,
-                                          STARTTIME, ENDTIME, attach_response=True)
+                                          STARTTIME - 2 * config.taper_val,
+                                          ENDTIME + 2 * config.taper_val,
+                                          attach_response=True)
             except header.FDSNNoDataException:
                 print(f"No data retrieved for {STA}")
                 continue
 
             st.merge(fill_value='latest')
-            st.trim(STARTTIME, ENDTIME, pad='true', fill_value=0)
+            st.trim(STARTTIME - 2 * config.taper_val, ENDTIME + 2 * config.taper_val, pad='true', fill_value=0)
             st.sort()
             print(st)
 
             print('Removing sensitivity...')
             st.remove_sensitivity()
 
             stf = st.copy()
+            stf.taper(max_percentage=None, max_length=config.taper_val)
```

Comment:
So if we are just going to trim the taper off, why do it at all? Why not just skip this step?

Reply:
A taper reduced artifical edge effects caused by filtering, but we don't want to plot a tapered trace. So basically the point here is to download data that extends before our desired STARTTIME and after our desired ENDTIME. Then we apply the taper outside those bounds. After we filter, we trim the trace to STARTTIME/ENDTIME, which removes the taper part.

Reply:
Ah, artificial edge effects. Gotcha, that makes sense.